tests/RestfulServerTest.php

```diff
@@ -0,0 +1,562 @@
+            $formatter->convertDataObject($editor, ["FirstName", "Email"]),
+            "Correct JSON formatting with field subset"
+        );
+
+        $set = Member::get()
+            ->filter('ID', [$editor->ID, $user->ID])
+            ->sort('"Email" ASC'); // for sorting for postgres
+        $this->assertEquals(
+            '{"totalSize":null,"items":[{"FirstName":"Editor","Email":"editor@test.com"},' .
+                '{"FirstName":"User","Email":"user@test.com"}]}',
+            $formatter->convertDataObjectSet($set, ["FirstName", "Email"]),
+            "Correct JSON formatting on a dataobjectset with field filter"
+        );
+    }
+
+    public function testApiAccessWithPOST()
+    {
+        $urlSafeClassname = $this->urlSafeClassname(RestfulServerTestAuthorRating::class);
+        $url = "{$this->baseURI}/api/v1/$urlSafeClassname/";
+        $data = [
+            'Rating' => '42',
+            'WriteProtectedField' => 'haxx0red'
+        ];
+        $response = Director::test($url, $data, null, 'POST');
+        // Assumption: XML is default output
+        $responseArr = Convert::xml2array($response->getBody());
+        $this->assertEquals(42, $responseArr['Rating']);
+        $this->assertNotEquals('haxx0red', $responseArr['WriteProtectedField']);
+    }
+
+    public function testCanViewRespectedInList()
+    {
+        // Default content type
+        $urlSafeClassname = $this->urlSafeClassname(RestfulServerTestSecretThing::class);
+        $url = "{$this->baseURI}/api/v1/$urlSafeClassname/";
+        $response = Director::test($url, null, null, 'GET');
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertNotContains('Unspeakable', $response->getBody());
+
+        // JSON content type
+        $url = "{$this->baseURI}/api/v1/$urlSafeClassname.json";
+        $response = Director::test($url, null, null, 'GET');
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertNotContains('Unspeakable', $response->getBody());
+        $responseArray = Convert::json2array($response->getBody());
+        $this->assertSame(0, $responseArray['totalSize']);
+
+        // With authentication
+        $_SERVER['PHP_AUTH_USER'] = 'editor@test.com';
+        $_SERVER['PHP_AUTH_PW'] = 'editor';
+        $urlSafeClassname = $this->urlSafeClassname(RestfulServerTestSecretThing::class);
+        $url = "{$this->baseURI}/api/v1/$urlSafeClassname/";
+        $response = Director::test($url, null, null, 'GET');
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertContains('Unspeakable', $response->getBody());
+        // Assumption: default formatter is XML
+        $responseArray = Convert::xml2array($response->getBody());
+        $this->assertEquals(1, $responseArray['@attributes']['totalSize']);
+        unset($_SERVER['PHP_AUTH_USER']);
+        unset($_SERVER['PHP_AUTH_PW']);
+    }
+}
```

Comment:
Now: Can you move these test classes back into tests/unit so git won't think they're replaced?

Reply:
done.